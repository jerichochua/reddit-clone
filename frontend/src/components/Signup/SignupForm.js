import React from 'react';
import { Form, FormField, FormButton } from '../Form';

const SignupForm = () => {
  return (
    <Form>
      <FormField label='username' type='text' placeholder='username' />
      <FormField label='password' type='password' placeholder='password' />
      <FormField
        label='confirm password'
        type='password'
        placeholder='confirm password'
      />
      <FormButton label='sign up' type='submit' />
    </Form>
  );
};

export default SignupForm;
