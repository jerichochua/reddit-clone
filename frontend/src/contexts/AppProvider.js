import React, { createContext, useContext, useReducer } from 'react';
import jwt_decode from 'jwt-decode';

const AppContext = createContext();
AppContext.displayName = 'AppContext';

const token = localStorage.getItem('token');
const userId = token ? jwt_decode(token).id : null;
const username = token ? jwt_decode(token).username : null;

const initialState = {
  userId,
  username,
  token,
};

const reducer = (state, action) => {
  switch (action.type) {
    case 'SET_USER':
      return {
        ...state,
        userId: action.payload.userId,
        username: action.payload.username,
      };
    case 'SET_TOKEN':
      return { ...state, token: action.payload };
    case 'LOGOUT':
      return { ...state, userId: null, username: null, token: null };
    default:
      return state;
  }
};

export const AppProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  return (
    <AppContext.Provider value={{ state, dispatch }}>
      {children}
    </AppContext.Provider>
  );
};

export const useAppContext = () => {
  return useContext(AppContext);
};
