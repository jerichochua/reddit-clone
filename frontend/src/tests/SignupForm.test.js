import renderer from 'react-test-renderer';
import { AppProvider } from '../contexts/AppProvider';
import SignupForm from '../components/Signup/SignupForm';

it('renders correctly', () => {
  const component = renderer.create(
    <AppProvider>
      <SignupForm />
    </AppProvider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
