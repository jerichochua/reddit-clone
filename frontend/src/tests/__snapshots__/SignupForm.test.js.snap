// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form-container"
  style={
    Object {
      "maxWidth": "375px",
    }
  }
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="form-field"
    >
      <label>
        username
      </label>
      <input
        name="username"
        onChange={[Function]}
        placeholder="username"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-field"
    >
      <label>
        password
      </label>
      <input
        name="password"
        onChange={[Function]}
        placeholder="password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-field"
    >
      <label>
        confirm password
      </label>
      <input
        name="confirmPassword"
        onChange={[Function]}
        placeholder="confirm password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <button
      className="form-btn"
      type="submit"
    >
      sign up
    </button>
  </form>
</div>
`;
